modules/javafx.controls/src/main/java/javafx/scene/control/ResizeFeaturesBase.java

```diff
@@ -96,7 +97,24 @@ public ResizeFeaturesBase(@NamedArg("column") TableColumnBase<S,?> column, @Name
    *
    * @since 20
    */
-  public void setColumnWidth(TableColumnBase<S,?> col, double width) {
-      col.doSetWidth(width);
+  public void setColumnWidth(TableColumnBase<S, ?> col, double width) {
+      Control c = getTableControl();
+      if (c.isSnapToPixel()) {
+          double min = c.snapSizeX(col.getMinWidth());
+          double max = RegionHelper.snapInnerSpaceX(c, col.getMaxWidth());
+          if (width < min) {
+              width = min;
+          } else if (width > max) {
+              width = max;
+              if (width < min) {
+                  // safety check in case floor(max) < ceil(min)
+                  width = min;
+              }
+          } else {
+              width = c.snapPositionX(width);
+          }
+      }
+      // can set width directly because all constraints have been checked
+      col.setWidth(width);
```

Comment:
The constraints have been checked only in the default snap-to-pixel case. This will bypass clamping if snap-to-pixel is false. I recommend moving the `setWidth` call into the if statement, and calling `doSetWidth` in an `else` block.